test/DebugInfo/nested_salvage_struct.swift

```diff
@@ -0,0 +1,19 @@
+// RUN: %target-swiftc_driver %s -emit-sil -g -O -o - | %FileCheck %s
+
+struct Wrapper {
+  let bytes: UInt64
+
+  private func pop(numBits: Int) -> Wrapper {
+    // `self` in here is a nested fragment. It should be salvaged, when inlined in `maker(numBits:)`
+    return Self(bytes: bytes &>> numBits)
+  }
+
+  // CHECK-LABEL: sil {{.*}} @$s4main7WrapperV5maker7numBitsACSi_tFZ
+  public static func maker(numBits: Int) -> Wrapper {
+    // CHECK: debug_value %0 : $Int, let, name "numBits", argno 1
+    // This function is static, but `pop`'s self should be inlined in here
+    // CHECK: debug_value %{{.*}} : $Builtin.Int64, let, name "self", {{.*}}, implicit, type $Wrapper, expr op_fragment:#Wrapper.bytes:op_fragment:#UInt64._value
+
```

Comment:
This is going to be a bit fragile, since it depends on the entire optimizer pipeline. Do you think you could extract a SIL testcase from this that just runs one pass?